### PR TITLE
[OAP-1771]Update ReadMe for better user experience in oap-data-source installation

### LINKED ISSUE #1771: [oap-data-source][arrow] Update README.md for how to build and use data source jar
The ReadMe has been out-of-dated and must be refreshed for OAP 0.9.
1. Remove libhdfs3.so
2. Support libhdfs.so
3. Some issues happen when following the installation guide.
3.1 CMake version must be 3.2 or higher(2.8.12 by default in CentOS7)
3.2 maven version must be 3.1.1 or higher(3.0.5 by default in CentOS7)
3.3 libgandiva_jni.so was not found inside JAR
3.4 Two failures when compiling intel-bigdata's arrow gandiva

### PR DESCRIPTION
**Pull request by weiting-chen**
## What changes were proposed in this pull request?
Closes #1771 
An updated ReadMe for oap-data-source installation and set up.

## How was this patch tested?
Only ReadMe update and has been tested offline step by step

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: oap-data-source/arrow/README.md [hunk 1]
```diff
@@ -10,27 +10,93 @@ We have provided a Conda package which will automatically install dependencies n
...
+
+// Use alternatives to choose mvn version
+sudo alternatives --config mvn
+```
+
+### Hadoop Native Library(Default)
+
+Please make sure you have set up Hadoop directory properly with Hadoop Native Libraries
+In your $HADOOP_HOME/lib/native directory, there are libhadoop.so and libhdfs.so existing.
+By default, arrow would scan your $HADOOP_HOME and find the native Hadoop library to be used for Hadoop connection.
```
Comment (zhouyuan) [inline oap-data-source/arrow/README.md:67]: this maybe one uncommon case, what will happen if hadoop nodes are remote?

File: oap-data-source/arrow/README.md [hunk 2]
```diff
@@ -10,27 +10,93 @@ We have provided a Conda package which will automatically install dependencies n
...
+
+### Use libhdfs3 library for better performance(Optional)
 
 For better performance ArrowDataSource reads HDFS files using the third-party library libhdfs3. The library should be pre-installed 
 on machines Spark Executor nodes are running on.
 
-To install the library, use of [Conda](https://docs.conda.io/en/latest/) is recommended.
+There are two ways to set up libhdfs3 library
+Option1: Overwrite soft link for libhdfs.so
+You can find libhdfs3.so in $OAP_DIR/oap-native-sql/cpp/src/resources/libhdfs3.so
```
Comment (zhouyuan) [inline oap-data-source/arrow/README.md:78]: maybe we should suggest to use libhdfs3.so from conda package

File: oap-data-source/arrow/README.md [hunk 3]
```diff
@@ -76,6 +148,27 @@ To enable ArrowDataSource, the previous built jar `spark-arrow-datasource-0.9.0-
...
+        --conf spark.executor.instances=12 \
+        --conf spark.executor.cores=6 \
+        --conf spark.executor.memory=20G \
+        --conf spark.memory.offHeap.size=80G \
+        --conf spark.task.cpus=1 \
+        --conf spark.locality.wait=0s \
+        --conf spark.sql.shuffle.partitions=72 \
+        --conf spark.executorEnv.ARROW_LIBHDFS3_DIR="$PATH_TO_LIBHDFS3_DIR/" \
+        --conf spark.executorEnv.LD_LIBRARY_PATH="$PATH_TO_LIBHDFS3_DEPENDENCIES_DIR" \
+        --jars ${NATIVE_SQL_DATASOURCE_JAR}
```
Comment (zhouyuan) [inline oap-data-source/arrow/README.md:169]: what is the jars for this case?

File: oap-data-source/arrow/README.md [hunk 4]
```diff
@@ -76,6 +148,27 @@ To enable ArrowDataSource, the previous built jar `spark-arrow-datasource-0.9.0-
...
+```
+
 For more information about these options, please read the official Spark [documentation](https://spark.apache.org/docs/latest/configuration.html#runtime-environment).
 
 ### Run a query (Scala)
```
Comment (zhouyuan) [inline oap-data-source/arrow/README.md:174]: is it possible to verify arrow data source is working, like showing a example DAG program picture?

**Comment by zhouyuan** [comment]
@weiting-chen thanks for the great efforts!

File: oap-data-source/arrow/README.md (hunk 1 above)
Comment (weiting-chen) [reply oap-data-source/arrow/README.md]: Done to add more description for this case.

File: oap-data-source/arrow/README.md (hunk 2 above)
Comment (weiting-chen) [reply oap-data-source/arrow/README.md]: Done to add the description for installing libhdfs3.so from conda package.

File: oap-data-source/arrow/README.md (hunk 3 above)
Comment (weiting-chen) [reply oap-data-source/arrow/README.md]: Did some testing for "--conf spark.executor.extraClassPath" and "--jar".
Both ways can support to add a jar to the executors.
Let's keep one solution(spark.executor.extraClassPath) to be clear for the user and remove --jar part.

File: oap-data-source/arrow/README.md (hunk 4 above)
Comment (weiting-chen) [reply oap-data-source/arrow/README.md]: Add a picture to show if arrow data source is enabled.

File: oap-data-source/arrow/README.md [hunk 5]
```diff
@@ -90,3 +194,9 @@ val df = spark.read
 df.createOrReplaceTempView("my_temp_view")
 spark.sql("SELECT * FROM my_temp_view LIMIT 10").show(10)
 ```
+
+### To validate if ArrowDataSource works properly
+
+To validate if ArrowDataSource works, you can go to the DAG to check if ArrowScan has been used from the above example query. 
+
+![Image of ArrowDataSource Validation](https://github.com/Intel-bigdata/OAP/tree/branch-0.9-spark-3.x/oap-data-source/arrow/resource/arrowdatasource_validation.png)
```
Comment (zhouyuan) [inline oap-data-source/arrow/README.md:202]: it seems this file `arrowdatasource_validation.png` is missing

Comment (weiting-chen) [reply oap-data-source/arrow/README.md]: Fixed. change the path to a relative path.

**Review by zhouyuan — APPROVED** [review]

**Comment by HongW2019** [comment]
@weiting-chen Do we also need to cherry-pick this commit to master?

**Comment by weiting-chen** [comment]
> @weiting-chen Do we also need to cherry-pick this commit to master?

Yes, but a couple of places must be changed for master, will cherry-pick from this commit with the modification, and send a new PR to address it in master.

**Pull request merged**